### DISCUSSION TITLE
Allow triggering external webhooks on events

### DISCUSSION THREAD

**Discussion by gw0** [grafana/grafana#38172] [Dashboards, Panels & Visualizations]
**What would you like to be added**:
Support for triggering external webhooks on specific events (e.g. when saving a dashboard). Probably for events that the backend already broadcasts for other purposes, e.g.:

https://github.com/grafana/grafana/blob/5fcff1a4c74f80fe5c29cc259d43418b35088467/pkg/api/dashboard.go#L266

**Why is this needed**:
In order to automate provisioning of dashboards it would be beneficial if Grafana could trigger an external webhook each time a dashboard is saved through the Web UI. This would enable an external service to know that a dashboard changed. The service could then use the Grafana API to retrieve the dashboard and open a PR to automate GitOps. For example, to automate authors workflow in https://github.com/grafana/grafana/issues/11778 even further.

Currently, something similar could be implemented with an external service with a pull strategy that is checking for changes in dashboards using the Grafana API. Constant checking introduces a lot of overhead, therefore implementing a push strategy with a webhook makes more sense.

**Comment by yesoreyeram** [comment]
Probably duplicate of #9066

**Comment by pkolyvas** [comment]
We're in the process of moving Grafana feature requests to [GitHub Discussions](https://community.grafana.com/t/were-moving-grafana-feature-requests-to-github-discussions/50341/3). This issue is being moved to [Discussions](https://github.com/grafana/grafana/discussions) or has been moved from our issues list, depending on where you're seeing this message. 

 We hope to use the discussion thread to continue building use cases, and additional evidence based on engagement/voting, in order to help us prioritize this feature request later on down the road. 

As with all things Grafana, we value your input and participation in this project and appreciate your patience during this transitional period. Looking forward to the discussion!

**Comment by gelicia** [comment]
Hello, as you may have heard, we are [transitioning away from using discussions to discuss feature requests](https://github.com/grafana/grafana/discussions/73424). Due to the age and number of responses to this discussion, we are deciding to close it. If this is something you would like to see in Grafana, feel free to open an [issue](https://github.com/grafana/grafana/issues) so the discussion can continue. Thank you!
